src/lib/swapper/swappers/OneInchSwapper/endpoints.ts

```diff
@@ -22,13 +26,37 @@ export const oneInchApi: Swapper2Api = {
     input: GetTradeQuoteInput,
     { sellAssetUsdRate }: { sellAssetUsdRate: string },
   ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
-    const tradeQuoteResult = await getTradeQuote(input as GetEvmTradeQuoteInput, sellAssetUsdRate)
+    const { sellAsset, sellAmountBeforeFeesCryptoBaseUnit, affiliateBps, receiveAddress } = input
 
+    const sellAmountBeforeFeesCryptoPrecision = fromBaseUnit(
+      sellAmountBeforeFeesCryptoBaseUnit,
+      sellAsset.precision,
+    )
+    const sellAmountBeforeFeesUsd = bnOrZero(sellAmountBeforeFeesCryptoPrecision).times(
+      sellAssetUsdRate,
+    )
+    // We use the sell amount so we don't have to make 2 network requests, as the receive amount requires a quote
+    const isDonationAmountBelowMinimum =
+      sellAmountBeforeFeesUsd.lt(MINIMUM_DONATION_Sell_AMOUNT_USD_FROM_ETH_CHAIN) &&
+      sellAsset.chainId === KnownChainIds.EthereumMainnet
```

Comment:
Obviously very nitpick but could have a mapping of chain to minimum donation amount similar to `MIN_COWSWAP_USD_TRADE_VALUES_BY_CHAIN_ID`, minimum being 0 for all other chains so `getTradeQuote()` isn't concerned by this

Reply:
Extracted to a helper in https://github.com/shapeshift/web/pull/4902/commits/fbab9f47bd76fb3900fa6603cad4ab2997ed20d4